### ISSUE TITLE
useDeferredValue hook updates the deferred value immediately instead of after background re-render

### ISSUE THREAD

**Issue by Pluto0104** [facebook/react#27141]
React version: 18.2.0

## Steps To Reproduce

1. Create a React app using create-react-app with React 18 and React Scripts 5.
2. Create a component that uses `useDeferredValue` hook to defer updating a part of the UI based on a state variable.
3. Render the component and type something in the input.

Link to code example: https://codesandbox.io/s/trusting-butterfly-lfg874?file=/App.js

## The current behavior
The deferred value changes immediately when the original value changes, instead of after React finishes the original re-render and starts working on a background re-render with the new value.

## The expected behavior
The deferred value should not change immediately when the original value changes, but only after React finishes the original re-render and starts working on a background re-render with the new value, as described in the React documentation: https://react.dev/reference/react/useDeferredValue#returns

**Comment by quiteeasy** [comment]
Actually, the differed value is not changing immediately. It is working as stated in the [docs.](https://react.dev/reference/react/useDeferredValue#returns) 

>  React will first attempt a re-render with the old value (so it will return the old value), and then try another re-render in background with the new value (so it will return the updated value).

In your example everything is working right as described in docs. There are not many operations happening in background in your example so after the first render with an old value the engine immediately performs another background re-render and after finishing the first render you see an old value and almost immediately a new deferred one as the next re-render happens.

You can have a look at my [code sample](https://codesandbox.io/s/nervous-thompson-5y96mf?file=/App.js). Please, open the console and pay attention to `isStale, query, deferredQuery` variables. When console shows `isStale` as true it means that the value of variable is being differed at the moment you can see values of each variable in console log at every moment of code execution.

**Comment by eps1lon** [comment]
@quiteeasy's answer is correct. We can commit immediately since nothing suspends with the `deferredQuery` value.
